javascript/javascript-import-and-require.md

```diff
@@ -0,0 +1,149 @@
+|You will have dynamic loading wherever the loaded module<br> name is not predefined.Loading is synchronous.<br> Meaning if you have got multiple requires,<br> they’re loaded and processed one by one.|Import is asynchronous and may perform a touch higher.<br> Also, the require module system is not<br> customarily based mostly. |
+
+**when we can and cannot use IMPORT in our Web Application?**
+
+The **import** statement is used to get bindings which are exported by another module and imported modules are in strict mode whether you declare them as such or not. There is one condition when **import** statement cannot be used in embedded scripts, till the time you delcare type="module" in script file.
+
+**why IMPORT and REQUIRE neither works in the browser unless you fake it with webpack or equivalent?**
+
+Webpack has resolver library, which help import and require function in resoluting the url path and also help in reading the file present in different folder. This library helps in locating a module by its absolute path and  helps webpack find the module code that needs to be included in the bundle for every such require/import statement. webpack uses enhanced-resolve to resolve file paths while bundling modules.
+
+Curently, There are 3 resolving rules in webpack,
+
+1. Absolute Path:
+      we already have root path and didn't need any resoultion.
+      
+        ```
+        import '/home/Abhishek/file';
+        import 'C:\\Users\\Abhishek\\file';
+        ```
+2. Relative paths:
+
+      Directory of the resource file where the import or require occurs is taken to be the context directory. The relative path                                                          specified in the import/require is joined to this context path to produce the absolute path to the module.
+      
+      ```
+      import '../src/config1';
+      import './config2';
+      ```
+3. Module paths:
+      Modules are searched for inside all directories specified in resolve.modules. You can replace the original module path by an alternate path by creating an alias for it using the resolve.alias configuration option.
+      
+      ```
+      import 'module';
+      import 'module/lib/config';
+      ```
+
+## Codes
+
+**Import code snippet:**
+
+**Get Module using EXPORT**
+
+dep.js
+
+```
+export{
+  var bar = "do you love coding?"
+  function foo(){
+      return "yes, It is my life.";
+  }
+}
+```
+
+**Get Module using IMPORT**
+
+```
+import {foo, bar} from “dep”;
+console.log(bar);
+foo();
+```
+
+**Require code snippet:**
+
+file.js
+```
+const Square = require('./square.js');
+const mySquare = new Square(2);
+console.log(`The area of mySquare is ${mySquare.area()}`);
+```
+
+square.js
+```
+module.exports = class Square {
+  constructor(width) {
+    this.width = width;
+  }
+
+  area() {
+    return this.width ** 2;
+  }
+};
+```
+
+## Challenge
+
+1. We will learn by doing and let us take example.

```

Comment:
What does this mean?

Reply:
Are these the directions for the lesson?

Please review the template to see what each section is for. https://github.com/Techtonica/curriculum/blob/master/_templates/lesson-v2.md

Reply:
If you are trying to give them directions, please be a lot more explicit about what they should do.  Should they make a new function in the same file, or a new one?  Tell them what the function should do, and what the files should be named. Then tell them which file they should import it into, and call it.  Do they know how to validate that it worked?